pkg/testgen/prow_yaml.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2022 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
```

Comment:
note: as with `generators.go`, not really sure what to call this file. `types.go` or `structs.go` work but they didn't really feel right. Again, open to suggestions or just leaving it as it doesn't really matter.

Reply:
`types.go` I think makes more sense after changing the other naming to `prow` + `job` where appropriate.